SW-1223 don't show the update failed info message on boot

JFSolution opened the pull request:
add user clicked update flag
redirect softwareupdateplugin check for update to our endpoint if the user clicked on it
refactor fallback update config for offline

Commit:
add user clicked update flag
redirect softwareupdateplugin check for update to our endpoint if the user clicked on it
refactor fallback update config for offline

Commit:
remove not needed debug output

ahmed-mrbeam commented on octoprint_mrbeam/__init__.py:
```diff
@@ -738,6 +739,15 @@ def calibration_tool_mode(self):
         return ret
 
+    @property
+    def user_checks_for_updates(self):
```
suggest the following refactor:
`user_checks_for_updates` --> `explicit_Update_Check`

JFSolution replied:
I can do this

ahmed-mrbeam commented on octoprint_mrbeam/__init__.py:
```diff
@@ -738,6 +739,15 @@ def calibration_tool_mode(self):
+        return self._user_checks_for_updates
+
+    @user_checks_for_updates.setter
+    def user_checks_for_updates(self, value):
```
Since it is a boolean value, I would suggest to create two setters, one for set and one for clear

JFSolution replied:
ok i can change that

JFSolution replied:
will change it to set_explicit_update_check and clear_explicit_update_check

ahmed-mrbeam commented on octoprint_mrbeam/rest_handler/update_handler.py:
```diff
@@ -12,5 +12,12 @@ class UpdateRestHandlerMixin:
+        clicked_by_user = False
+        if hasattr(request, "json") and request.json:
+            data = request.json
+            clicked_by_user = data.get("user", False)
```
Are you sure you want to default to False ?

JFSolution replied:
Yes, if it is not set the default should be false, as we only set it to true if the users clicks on it and otherwise leave this data element empty

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -54,6 +54,30 @@ class SWUpdateTier(Enum):
 BEAMOS_LEGACY_VERSION = "0.14.0"
 BEAMOS_LEGACY_DATE = date(2018, 1, 12)  # still used in the migrations
 
+FALLBACK_UPDATE_CONFIG = {
```
Is this change related to the fix ?

JFSolution replied:
Yes I needed to refactor this so I am still able to give back the fallback update config

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -168,32 +193,7 @@ def get_update_information(plugin):
-                },
-            },
-        },
+        {},
```
I don't understand what happened here

JFSolution replied:
this is releated to the refactoring https://github.com/mrbeam/MrBeamPlugin/pull/1463/files#r859892304
I refactored it so the _set_info_from_cloud_config method will return the `FALLBACK_UPDATE_CONFIG` if something is not correct instead of returning it only if there was a connection problem

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -229,14 +229,15 @@ def switch_software_channel(plugin, channel):
     Returns:
         None
     """
+    _logger.debug("switch_software_channel")
```
Could you please add at the begging of the log something  like "Software update information: "
so that it can be clear in the logs?

JFSolution replied:
this will be added from the logger it selfe as it logs everything from this file as `octoprint.plugins.mrbeam.software_update_information`

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -229,14 +229,15 @@ def switch_software_channel(plugin, channel):
 
 
-def reload_update_info(plugin):
+def reload_update_info(plugin, clicked_by_user=False):
```
Not sure why we have `clicked_by_user` and _user_checks_for_updates why not use one of them ?

JFSolution replied:
clicked by user is only the input parameter of reload_update_info, that will then be saved to the user_checks_for_updates

ahmed-mrbeam reviewed (commented):
Partially reviewed

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -245,6 +246,9 @@ def reload_update_info(plugin):
         None
     """
+    if clicked_by_user:
+        plugin.user_checks_for_updates = True
```
please use the setter

ahmed-mrbeam commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -320,17 +325,20 @@ def _set_info_from_cloud_config(plugin, tier, beamos_version, cloud_config):
                 f.write(json.dumps(sw_update_config))
         except (IOError, TypeError):
             plugin._logger.error("can't create update info file")
```
Please add here that Update fallback config will be used

ahmed-mrbeam commented on octoprint_mrbeam/static/js/software_channel_selector.js:
```diff
@@ -95,11 +95,12 @@ $(function () {
-            self.reload_update_info();
             if (force !== undefined) {
                 force = true; //only forces the update check if it was disabled ("check for update" button press)
+                self.reload_update_info(showIfNothingNew, force, ignoreSeen, true); //if use clicked update forward to our reload endpoint
```
`showIfNothingNew` `ignoreSeen`   why using different naming convention here ?

ahmed-mrbeam commented on tests/softwareupdate/test_cloud_config.py:
```diff
@@ -461,10 +438,11 @@ def test_cloud_confg_fileerror(
                 },
             )
             plugin = self.plugin
+            plugin.user_checks_for_updates = True
```
Please use the setter

Commit:
added feedback of PR
- refactored user_checks_for_updates to explicit_update_check
- refactored the notifications to use a function to follow DRY
- extended logging to include errorcodes
- added testcase for E-1003 error message
- fixed testcases to work with refactoring

Commit:
fix missing spinner during update check
